File: src/main/java/com/claudio_personagens/bookingcharactersapi/base/AbstractRepository.java
package com.claudio_personagens.bookingcharactersapi.base;

import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.repository.NoRepositoryBean;

import java.io.Serializable;

@NoRepositoryBean
public interface AbstractRepository<T, ID extends Serializable> extends JpaRepository<T, ID> {

}

File: src/main/java/com/claudio_personagens/bookingcharactersapi/base/AbstractRepository.kt
package com.claudio_personagens.bookingcharactersapi.base

import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.data.repository.NoRepositoryBean
import java.io.Serializable

@NoRepositoryBean
interface AbstractRepository<T, ID : Serializable?> : JpaRepository<T, ID>